### PR TITLE
Benchmarker and aggregator: make it easier to find and use them

### PR DESCRIPTION
**Pull request by ge0ffrey**
Feedback from Boris. Especially the javadoc link from CloudBalancingHelloWorld to CloudBalancingBenchmarkHelloWorld is crucial.

### CODE REVIEW

File: optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/api/PlannerBenchmarkFactory.java [hunk 1]
```diff
@@ -131,7 +131,7 @@ public static PlannerBenchmarkFactory createFromKieContainerXmlResource(ReleaseI
      * @return never null
      */
     public static PlannerBenchmarkFactory createFromKieContainerXmlResource(KieContainer kieContainer,
-                                                                            String benchmarkConfigResource) {
+            String benchmarkConfigResource) {
```
Comment (triceo) [inline optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/api/PlannerBenchmarkFactory.java]: Is this the proper formatting of multi-line arguments? If so, I have to reconfigure my IDE.

File: optaplanner-examples/src/main/java/org/optaplanner/examples/cloudbalancing/optional/benchmark/CloudBalancingBenchmarkHelloWorld.java [hunk 2]
```diff
@@ -16,19 +16,68 @@
...
 
     public static void main(String[] args) {
+        List<String> argList = Arrays.asList(args);
+        boolean advanced = argList.contains("--advanced");
+        if (!advanced) {
+            runBasicBenchmark();
+        } else {
+            boolean aggregator = argList.contains("--aggregator");
+            runAdvancedBenchmark(aggregator);
+        }
```
Comment (triceo) [inline optaplanner-examples/src/main/java/org/optaplanner/examples/cloudbalancing/optional/benchmark/CloudBalancingBenchmarkHelloWorld.java]: Shouldn't this behavior be somehow shared across all the individual benchmark apps that can be found in the examples?

File: optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/api/PlannerBenchmarkFactory.java (hunk 1 above)
Comment (ge0ffrey) [reply optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/api/PlannerBenchmarkFactory.java]: It's an open, painful discussion. At some point it was 2 indents in the kie standard - and it was in sync with the old Java Sun standards and Jboss standards - but then someone changed it to align on column - going out of sync with most of the broader Java community.

Honestly, align on columns sucks big time, especially for files with advanced generics, so in the optaplanner repo, we're doing 2 indents... Let's talk about it some time.

File: optaplanner-examples/src/main/java/org/optaplanner/examples/cloudbalancing/optional/benchmark/CloudBalancingBenchmarkHelloWorld.java (hunk 2 above)
Comment (ge0ffrey) [reply optaplanner-examples/src/main/java/org/optaplanner/examples/cloudbalancing/optional/benchmark/CloudBalancingBenchmarkHelloWorld.java]: This is the only hello world class for a benchmark - all the BenchmarkApp classes use the "examples common framework".

**Comment by triceo** [comment]
@ge0ffrey With the freeze upon us, shouldn't we merge this?

**Pull request merged**